### PR TITLE
docs(feature_flags): snippets split, improved, and lint

### LINKED ISSUE #1373: Refactor Feature Flags code snippets

### PR DESCRIPTION
**Pull request by leandrodamascena**
**Issue number:** #1373 

## Summary

Changes snippet fictitious names to real code snippet filenames for easier discovery and collaboration to improve/correct document snippets.

Some examples have also been refactored so that the codes are complete and there are no lint/pre-commit hooks issues.

### User experience

**BEFORE**
image

**AFTER**
image

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by rubenfonseca** [comment]
Will review this later today

File: docs/utilities/feature_flags.md [hunk 1]
```diff
@@ -24,94 +24,37 @@ Feature flags are used to modify behaviour without changing the application's co
...
-    AWS AppConfig requires two API calls to fetch configuration for the first time. You can improve latency by consolidating your feature settings in a single [Configuration](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-configuration-and-profile.html).
+    AWS AppConfig requires two API calls to fetch configuration for the first time. You can improve latency by consolidating your feature settings in a single [Configuration](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-creating-configuration-and-profile.html){target="_blank"}.
 
 ## Key features
 
 * Define simple feature flags to dynamically decide when to enable a feature
 * Fetch one or all feature flags enabled for a given application context
 * Support for static feature flags to simply turn on/off a feature without rules
+* Support for time based feature flags
+* Bring Your Own Feature Flags Store Provider
```
Comment (rubenfonseca) [inline docs/utilities/feature_flags.md]:
```suggestion
* Bring your own Feature Flags store provider
```
Just to make it consistent with the other bullet points?

File: docs/utilities/feature_flags.md [hunk 2]
```diff
@@ -752,45 +446,65 @@ These are the available options for further customization.
...
+=== "appconfig_provider_options_features.json"
+
+    ```json hl_lines="6-9"
+    --8<-- "examples/feature_flags/src/appconfig_provider_options_features.json"
+    ```
+
+### Create your own store provider
 
-import jmespath
+You can create your own custom FeatureFlags store provider by inheriting the `StoreProvider` class, and implementing both `get_raw_configuration()` and `get_configuration()` methods to retrieve the configuration from your custom store.
```
Comment (rubenfonseca) [inline docs/utilities/feature_flags.md:478]: When reading the text and the sample, I don't understand what's the difference between the two methods `get_raw_configuration` and `get_configuration`. Do you think we can expand a little bit on that?

**Review by rubenfonseca — CHANGES_REQUESTED** [review]
I pushed two small fixes and left two comments to think about. Great great work Leandro!

File: docs/utilities/feature_flags.md (hunk 1 above)
Comment (leandrodamascena) [reply docs/utilities/feature_flags.md]: Done

File: docs/utilities/feature_flags.md (hunk 2 above)
Comment (leandrodamascena) [reply docs/utilities/feature_flags.md:478]: In fact, we need to explain this in a little more detail, just as you did in batch processing.

I did this, can you check pls?

**Review by rubenfonseca — APPROVED** [review]
Loved the changes explaining the `get` vs `get_raw`. Thank you for this awesome work Leo!

**Pull request merged**